### PR TITLE
feat: button group component design

### PR DESCRIPTION
**Pull request by volkangurenekli**

### CODE REVIEW

File: src/app/shared/components/button-group/button-group.component.ts [hunk 1]
```diff
@@ -1,12 +1,49 @@
...
+    private renderer: Renderer2,
+  ) {}
+
+  ngOnInit(): void {
+    this.options = {
+      ...this.defaultOptions,
+      ...this.options,
+    };
+
+    this.renderer.listen(this.elRef.nativeElement, 'click', (e) => this.click.emit(e));
```
Comment (ahmetatar) [inline src/app/shared/components/button-group/button-group.component.ts:47]: Why did you use a renderer here? You can make event binding to an element in the template.

File: src/app/shared/components/button-group/button-group.stories.ts [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+    click: actions.onClick,
+  },
+  template: `
+    <desx-button-group
+      [id]="id"
+      [btnGroupStyle]="style"
+      [options]="options"
+      (click)="onClick($event)"
+    >
+    <button type="button" class="btn btn-primary">Left</button>
```
Comment (ahmetatar) [inline src/app/shared/components/button-group/button-group.stories.ts:52]: How do these work with the desx-buttons?

File: src/app/shared/components/button-group/button-group.stories.ts [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+  `,
+});
+
+/**
+ * Default story
+ */
+export const Default = Template.bind({});
+Default.args = {
+  id: 'buttonGroup',
+  options: {
```
Comment (ahmetatar) [inline src/app/shared/components/button-group/button-group.stories.ts:65]: Isn't the default value already false?

File: src/app/shared/components/button-group/pipes/button-group-style.pipe.spec.ts [hunk 4]
```diff
@@ -0,0 +1,14 @@
+import { ButtonGroupOptions } from '../button-group.options';
+import { ButtonGroupStylePipe } from './button-group-style.pipe';
+
+describe('ButtonGroupStylePipe', () => {
+  it('transforms button class name by outline option', () => {
```
Comment (ahmetatar) [inline src/app/shared/components/button-group/pipes/button-group-style.pipe.spec.ts:5]: Don't leave the evidence lying around when Copy/Paste ;)

**Review by ahmetatar — CHANGES_REQUESTED** [review]
Looks very good overall but some changes are needed

File: src/app/shared/components/button-group/button-group.component.ts (hunk 1 above)
Comment (volkangurenekli) [reply src/app/shared/components/button-group/button-group.component.ts]: renderer removed

File: src/app/shared/components/button-group/button-group.stories.ts (hunk 2 above)
Comment (volkangurenekli) [reply src/app/shared/components/button-group/button-group.stories.ts]: buttons changed

File: src/app/shared/components/button-group/pipes/button-group-style.pipe.spec.ts (hunk 4 above)
Comment (volkangurenekli) [reply src/app/shared/components/button-group/pipes/button-group-style.pipe.spec.ts]: every crime leaves a trace :)

File: src/app/shared/components/button-group/button-group.component.html [hunk 5]
```diff
@@ -1 +1,3 @@
-<p>button-group works!</p>
+<div [ngClass]="[btnGroupStyle | buttonGroupStyle: options, size]">
+  <ng-content></ng-content>
```
Comment (ahmetatar) [inline src/app/shared/components/button-group/button-group.component.html:2]: What can be inside the button group other than the button? If this is an ng-content I can give the button group any component as content. Does your view support this?

File: src/app/shared/components/button-group/button-group.component.ts [hunk 6]
```diff
@@ -1,12 +1,29 @@
...
+  encapsulation: ViewEncapsulation.None,
 })
 export class ButtonGroupComponent implements OnInit {
-  constructor() {}
+  @Input()
+  options?: Partial<ButtonGroupOptions>;
 
-  ngOnInit(): void {}
+  @Input()
+  size: ButtonGroupSizes = 'btn-lg';
```
Comment (ahmetatar) [inline src/app/shared/components/button-group/button-group.component.ts:16]: I'm concerned about the Bootstrap class names getting scattered around the component scope. Can you start a discussion about it?

**Review by ahmetatar — COMMENTED** [review]
Let's discuss for some changes

File: src/app/shared/components/button-group/button-group.component.html (hunk 5 above)
Comment (volkangurenekli) [reply src/app/shared/components/button-group/button-group.component.html]: conditional render added

File: src/app/shared/components/button-group/button-group.component.ts (hunk 6 above)
Comment (volkangurenekli) [reply src/app/shared/components/button-group/button-group.component.ts]: button group model updated

**Pull request closed without merging**